Detectors/Base/src/GRPGeomHelper.cxx

```diff
@@ -225,6 +241,36 @@ void GRPGeomHelper::checkUpdates(ProcessingContext& pc) const
         }
       }
     }
+    if (mRequest->askAggregateRunInfo) {
+      const auto hmap = pc.inputs().get<o2::framework::CCDBMetadataExtractor>("RCTRunInfo"); // metadata only!
+      mAggregatedRunInfo.runNumber = pc.services().get<o2::framework::TimingInfo>().runNumber;
+      auto rl = o2::ccdb::BasicCCDBManager::getRunDuration(hmap);
+      mAggregatedRunInfo.sor = rl.first;
+      mAggregatedRunInfo.eor = rl.second;
+      mAggregatedRunInfo.orbitsPerTF = getNHBFPerTF();
+      mAggregatedRunInfo.orbitReset = mOrbitResetTimeMUS;
+
+      mAggregatedRunInfo.orbitSOR = (mAggregatedRunInfo.sor * 1000 - mOrbitResetTimeMUS) / o2::constants::lhc::LHCOrbitMUS;
+      mAggregatedRunInfo.orbitEOR = (mAggregatedRunInfo.eor * 1000 - mOrbitResetTimeMUS) / o2::constants::lhc::LHCOrbitMUS;
+      // adjust to the nearest TF edge to satisfy condition (orbitSOR % nOrbitsPerTF == 0)
+      mAggregatedRunInfo.orbitSOR = (mAggregatedRunInfo.orbitSOR / mAggregatedRunInfo.orbitsPerTF + 1) * mAggregatedRunInfo.orbitsPerTF; // +1 to choose the safe boundary ... towards run middle
+      mAggregatedRunInfo.orbitEOR = mAggregatedRunInfo.orbitEOR / mAggregatedRunInfo.orbitsPerTF * mAggregatedRunInfo.orbitsPerTF;
```

Comment:
I do not understand this part. It seems that the calculation of  of `AggregateRunInfo::buildAggregateRunInfo` is repeated. Could this be avoided (by potentially refactoring methods in AggregatedRunInfo) ?